tedana/decomposition/gift_pca.py

```diff
@@ -0,0 +1,599 @@
+"""
+PCA based on GIFT software
+"""
+import logging
+
+import numpy as np
+from sklearn.decomposition import PCA, TruncatedSVD
+from sklearn.preprocessing import StandardScaler
+
+from scipy.signal import detrend
+from scipy.fftpack import fft, fftshift, fftn, fft2
+from scipy.signal import correlate2d
+
+LGR = logging.getLogger(__name__)
+
+
+def _autocorr(x):
+    """
+    Run Singular Value Decomposition (SVD) on input data,
+    automatically select components based on the GIFT software.
+
+    Parameters
+    ----------
+    x :
+
+    Returns
+    -------
+    u :
+    """
+    result = np.correlate(x, x, mode='full')
+    return result[result.size / 2:]
+
+
+def _sumN(dat):
```

Comment:
This is a one-line function and only used twice (both times in `_entrate_sp`). Let's just fold it in there.

Reply:
Should be done in the next commit.